### PR TITLE
78 and 79 - updated docBlocks, converted var to let, cleaned up spacing issues, introduced ES6 structures where possible and converted typeof to Ember.typeOf.

### PR DESCRIPTION
**Pull request by juwara0**
Issues: 78 and 79
  Updated docBlocks to follow javascript coding style guidelines.
  Converted var declarations to let declarations.
  Cleaned up spacing issues to follow javascript coding style guidelines.
  Introduced ES6 structures/syntax where appropriate.
  Converted usage of typeof to Ember.typeOf().

### CODE REVIEW

File: test-support/helpers/sl/synchronous/ajax.js [hunk 1]
```diff
@@ -1,39 +1,51 @@
...
+                Ember.$( document ).trigger( 'ajaxStart' );
+            }
+        });
+    }
+
+    /**
+     * Emulate the conclusion of an AJAX request
+     *
+     * @function
+     * @param {String} endpoint
```
Comment (notmessenger) [inline test-support/helpers/sl/synchronous/ajax.js]: parameter name should be in brackets to denote it is optional

File: test-support/helpers/sl/synchronous/ajax.js [hunk 2]
```diff
@@ -1,39 +1,51 @@
...
-        }
-    });
-};
+class AjaxHelper {
+
+    /**
+     * Emulate the beginning of an AJAX request
+     *
+     * @function
+     * @param {String} endpoint
```
Comment (notmessenger) [inline test-support/helpers/sl/synchronous/ajax.js]: parameter name should be in brackets to denote it is optional

File: test-support/helpers/sl/synchronous/contains.js [hunk 3]
```diff
@@ -13,25 +13,25 @@ import {
...
  * @function contains
- * @param   {Ember.Application} app (optional)
- * @param   {array|string|object} underTest
- * @param   {mixed} testFor
- * @throws  {Ember.assert}
- * @returns {boolean}
+ * @param {ember/Application} [app]
+ * @param {Array|String|Object} underTest
+ * @param {Array|String|Object} testFor
+ * @throws {ember/Assert}
```
Comment (notmessenger) [inline test-support/helpers/sl/synchronous/contains.js]: ember.assert

File: test-support/helpers/sl/synchronous/contains.js [hunk 4]
```diff
@@ -13,25 +13,25 @@ import {
...
  *
  * @function contains
- * @param   {Ember.Application} app (optional)
- * @param   {array|string|object} underTest
- * @param   {mixed} testFor
- * @throws  {Ember.assert}
- * @returns {boolean}
+ * @param {ember/Application} [app]
+ * @param {Array|String|Object} underTest
+ * @param {Array|String|Object} testFor
```
Comment (notmessenger) [inline test-support/helpers/sl/synchronous/contains.js]: Are Numbers allowed?

File: test-support/helpers/sl/synchronous/requires.js [hunk 5]
```diff
@@ -3,66 +3,66 @@ import Ember from 'ember';
...
+            required: false,
+            testValue: true,
+            message: 'Parameter was a boolean'
+        }
+    };
+    let testsThatHaveFailed = [];
+    let assertionThrown;
+    let assertionPassed;
+    let property;
+    let parameter;
```
Comment (notmessenger) [inline test-support/helpers/sl/synchronous/requires.js]: Move these let statements to a location closets to where they are being used, respective of the scope they require.

File: test-support/helpers/sl/utils/utils.js [hunk 6]
```diff
@@ -3,30 +3,30 @@ import Ember from 'ember';
 /**
  * Converts provided parameter into an array
  *
- * @function convertToArray
- * @param   {array|string|object} underTest
- * @throws  {Ember.assert} If not provided an Array, String or Object
- * @returns {array}
+ * @function
+ * @param {Array|String|Object} underTest
+ * @throws {Ember.assert} If not provided an Array, String or Object
```
Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: ember.assert

File: test-support/helpers/sl/utils/utils.js [hunk 7]
```diff
@@ -3,30 +3,30 @@ import Ember from 'ember';
...
 
     } else {
-        switch( typeof underTest ) {
+        switch( Ember.typeOf( underTest ) ) {
             case 'string':
-                    returnArray = convertStringToArray( underTest );
+                returnArray = convertStringToArray( underTest );
                 break;
 
             case 'object':
```
Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: there are more object types beyond this one now that you have switched to using Ember.typeOf

File: test-support/helpers/sl/utils/utils.js [hunk 8]
```diff
@@ -3,30 +3,30 @@ import Ember from 'ember';
...
-                    returnArray = convertObjectKeysToArray( underTest );
+                returnArray = convertObjectKeysToArray( underTest );
                 break;
         }
     }
 
-    Ember.assert( 'String, Object or Array must be supplied', 'undefined' !== typeof returnArray );
+    Ember.assert( 'String, Object or Array must be supplied', 'undefined' !== Ember.typeOf( returnArray ) );
```
Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: should be multi-line

File: test-support/helpers/sl/synchronous/contains.js (hunk 4 above)
Comment (juwara0) [inline test-support/helpers/sl/synchronous/contains.js]: No, only Object, String or Array

File: test-support/helpers/sl/utils/utils.js [hunk 9]
```diff
@@ -36,14 +36,14 @@ var convertToArray = function( underTest ) {
  *
  * Primarily exists to convert format of call to .prop( 'class' )
  *
- * @function convertStringToArray
- * @param   {string} underTest
- * @throws  {Ember.assert} If argument is not provided or is not a string
- * @returns {array}
+ * @function
+ * @param {String} underTest
+ * @throws {Ember.assert} If argument is not provided or is not a string
```
Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: ember.assert

File: test-support/helpers/sl/utils/utils.js [hunk 10]
```diff
@@ -36,14 +36,14 @@ var convertToArray = function( underTest ) {
...
+ * @returns {Array}
  */
-var convertStringToArray = function( underTest ) {
+let convertStringToArray = function( underTest ) {
 
-    Ember.assert( 'String must be supplied', 'string' === typeof underTest );
+    Ember.assert( 'String must be supplied', 'string' === Ember.typeOf( underTest ) );
```
Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: should be multi-line

File: test-support/helpers/sl/utils/utils.js [hunk 11]
```diff
@@ -53,27 +53,27 @@ var convertStringToArray = function( underTest ) {
  *
  * Property names are only extracted from the object provided.  No recursion into nested objects occurs.
  *
- * @function convertObjectKeysToArray
- * @param   {object} underTest
- * @throws  {Ember.assert} If argument is not provided or is not an object
- * @returns {array}
+ * @function
+ * @param {Object} underTest
+ * @throws {Ember.assert} If argument is not provided or is not an object
```
Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: ember.assert

File: test-support/helpers/sl/utils/utils.js [hunk 12]
```diff
@@ -53,27 +53,27 @@ var convertStringToArray = function( underTest ) {
...
+ * @returns {Array}
  */
-var convertObjectKeysToArray = function( underTest ) {
+let convertObjectKeysToArray = function( underTest ) {
 
-    Ember.assert( 'Object must be supplied', 'object' === typeof underTest && !Array.isArray( underTest ) );
+    Ember.assert( 'Object must be supplied', 'object' === Ember.typeOf( underTest ) && !Array.isArray( underTest ) );
```
Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: Should be multi-line

Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: there are more object types than just object that typeOf returns now that using it

Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: Should probably use typeOf for array for consistency

File: test-support/helpers/sl/synchronous/contains.js [hunk 13]
```diff
@@ -13,25 +13,25 @@ import {
...
-        underTest = arguments[index],
-        testFor   = arguments[index+1];
+    let index = ( 3 === arguments.length ) ? 1 : 0;
+    let underTest = arguments[ index ];
+    let testFor = arguments[ index + 1 ];
 
     Ember.assert(
         'First non-optional argument must be an array, string or object',
-        'object' === typeof underTest || 'string' === typeof underTest || Array.isArray( underTest )
+        'object' === Ember.typeOf( underTest ) || 'string' === Ember.typeOf( underTest ) || Array.isArray( underTest )
```
Comment (notmessenger) [inline test-support/helpers/sl/synchronous/contains.js]: Should probably use typeOf for array for consistency

File: test-support/helpers/sl/synchronous/contains.js [hunk 14]
```diff
@@ -13,25 +13,25 @@ import {
...
     );
 
     Ember.assert(
         'Second non-optional argument must be an array, string or object',
-        'object' === typeof testFor || 'string' === typeof testFor || Array.isArray( testFor )
+        'object' === Ember.typeOf( testFor ) || 'string' === Ember.typeOf( testFor ) || Array.isArray( testFor )
```
Comment (notmessenger) [inline test-support/helpers/sl/synchronous/contains.js]: Should probably use typeOf for array for consistency

File: test-support/helpers/sl/synchronous/contains.js (hunk 13 above)
Comment (notmessenger) [inline test-support/helpers/sl/synchronous/contains.js]: there are more object types beyond this one now that you have switched to using Ember.typeOf

File: test-support/helpers/sl/synchronous/contains.js (hunk 14 above)
Comment (notmessenger) [inline test-support/helpers/sl/synchronous/contains.js]: there are more object types beyond this one now that you have switched to using Ember.typeOf

File: test-support/helpers/sl/utils/utils.js [hunk 15]
```diff
@@ -53,27 +53,27 @@ var convertStringToArray = function( underTest ) {
...
- * @return {boolean}
+ * @function
+ * @param {Array} underTest
+ * @param {Array} testFor
+ * @returns {Boolean}
  */
-var doArraysIntersect = function( underTest, testFor ) {
+let doArraysIntersect = function( underTest, testFor ) {
 
     Ember.assert( 'Parameters must be Arrays', ( Array.isArray( underTest ) && Array.isArray( testFor ) ) );
```
Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: Should probably use typeOf for array for consistency

Comment (notmessenger) [inline test-support/helpers/sl/utils/utils.js]: Should be multi-line

File: tests/dummy/app/app.js [hunk 16]
```diff
@@ -3,7 +3,7 @@ import Resolver from 'ember/resolver';
 import loadInitializers from 'ember/load-initializers';
 import config from './config/environment';
 
-var App;
+let App;
 
 Ember.MODEL_FACTORY_INJECTIONS = true;
```
Comment (notmessenger) [inline tests/dummy/app/app.js]: The rest of this file should following spacing and formatting standards.  Change one change them all :-)

File: tests/helpers/resolver.js [hunk 17]
```diff
@@ -1,7 +1,7 @@
 import Resolver from 'ember/resolver';
 import config from '../../config/environment';
 
-var resolver = Resolver.create();
+let resolver = Resolver.create();
 
 resolver.namespace = {
   modulePrefix: config.modulePrefix,
```
Comment (notmessenger) [inline tests/helpers/resolver.js]: The rest of this file should following spacing and formatting standards. Change one change them all :-)

File: tests/helpers/start-app.js [hunk 18]
```diff
@@ -3,14 +3,14 @@ import Application from '../../app';
...
+  let attributes = Ember.merge( {}, config.APP );
+  attributes = Ember.merge( attributes, attrs ); // use defaults, but you can override;
 
-  Ember.run(function() {
-    application = Application.create(attributes);
+  Ember.run( function() {
+    application = Application.create( attributes );
     application.setupForTesting();
     application.injectTestHelpers();
   });
```
Comment (notmessenger) [inline tests/helpers/start-app.js]: The rest of this file should following spacing and formatting standards. Change one change them all :-)

File: tests/unit/helpers/sl/synchronous/requires-test.js [hunk 19]
```diff
@@ -171,8 +171,8 @@ test( 'Second argument must be an array', function( assert ) {
 });
 
 test( 'Return type', function( assert ) {
-    var testFunction = function( first ) {
-        Ember.assert( 'Test argument must be a function or boolean', 'function' === typeof first || 'boolean' === typeof first );
+    let testFunction = function( first ) {
+        Ember.assert( 'Test argument must be a function or boolean', 'function' === Ember.typeOf( first ) || 'boolean' === Ember.typeOf( first ) );
```
Comment (notmessenger) [inline tests/unit/helpers/sl/synchronous/requires-test.js]: should be multi-line

File: tests/unit/helpers/sl/synchronous/requires-test.js [hunk 20]
```diff
@@ -180,8 +180,8 @@ test( 'Return type', function( assert ) {
 });
 
 test( 'Functions as expected', function( assert ) {
-    var testFunction = function( first ) {
-        Ember.assert( 'Test argument must be a function or boolean', 'function' === typeof first || 'boolean' === typeof first );
+    let testFunction = function( first ) {
+        Ember.assert( 'Test argument must be a function or boolean', 'function' === Ember.typeOf( first ) || 'boolean' === Ember.typeOf( first ) );
```
Comment (notmessenger) [inline tests/unit/helpers/sl/synchronous/requires-test.js]: should be multi-line

File: tests/unit/helpers/sl/synchronous/ajax-test.js [hunk 21]
```diff
@@ -8,43 +9,43 @@ test( 'it exists', function( assert ) {
 });
 
 test( 'begin() with no parameter triggers "ajaxStart" event on document', function( assert ) {
-    var spy = sinon.spy( $.prototype, 'trigger' );
+    let spy = sinon.spy( Ember.$.prototype, 'trigger' );
 
     ajax.begin();
 
-    assert.equal( spy.args[0][0], 'ajaxStart' );
+    assert.equal( spy.args[ 0 ][ 0 ], 'ajaxStart' );
```
Comment (notmessenger) [inline tests/unit/helpers/sl/synchronous/ajax-test.js]: This, and all other asserts, should be multi-line

File: tests/unit/helpers/sl/synchronous/contains-test.js [hunk 22]
```diff
@@ -193,7 +193,7 @@ test( 'Arguments are passed to doArraysIntersect() in the correct order', functi
 });
 
 test( 'Returns a boolean', function( assert ) {
-    var response;
+    let response;
 
     response = contains( 'b', [ 'd', 'e' ] );
```
Comment (notmessenger) [inline tests/unit/helpers/sl/synchronous/contains-test.js]: All asserts should be multi-line

File: tests/unit/helpers/sl/synchronous/requires-test.js [hunk 23]
```diff
@@ -180,8 +180,8 @@ test( 'Return type', function( assert ) {
...
     },
     test;
```
Comment (notmessenger) [inline tests/unit/helpers/sl/synchronous/requires-test.js]: All asserts should be multi-line

File: tests/unit/helpers/sl/utils/utils-test.js [hunk 24]
```diff
@@ -26,15 +26,15 @@ test( 'doArraysIntersect() exists', function( assert ) {
 });
 
 test( 'convertToArray() requires either an Array, String, or Object to be provided', function( assert ) {
-    var test = requires( convertToArray, [ 'array', 'string', 'object' ] );
+    let test = requires( convertToArray, [ 'array', 'string', 'object' ] );
 
     assert.ok ( test.requires, test.messages );
```
Comment (notmessenger) [inline tests/unit/helpers/sl/utils/utils-test.js]: This, and all other asserts, should be multi-line

File: tests/unit/helpers/sl/utils/utils-test.js [hunk 25]
```diff
@@ -47,7 +47,7 @@ test( 'convertToArray() returns expected result', function( assert ) {
 });
 
 test( 'convertStringToArray() requires a string to be provided', function( assert ) {
-    var test = requires( convertStringToArray, [ 'string' ] );
+    let test = requires( convertStringToArray, [ 'string' ] );
 
     assert.ok ( test.requires, test.messages );
```
Comment (notmessenger) [inline tests/unit/helpers/sl/utils/utils-test.js]: This, and all other asserts, should be multi-line

**Pull request merged**